packages/remote-view/src/utils/getUrls.ts

```diff
@@ -0,0 +1,40 @@
+// Strip trailing slashes
+function normalizeBaseUrl(baseUrl: string) {
+  if (baseUrl.endsWith('/')) {
+    return baseUrl.slice(0, baseUrl.length - 1);
+  }
+
+  return baseUrl;
+}
+
+// Safe path to package.json
+export function getRemotePackageJsonUrl(baseUrl: string) {
+  const normalizedBaseUrl = normalizeBaseUrl(baseUrl);
+  return `${normalizedBaseUrl}/package.json`;
+}
+
+// Safe path to `module` or `style` defined in a package.json
+export function getRemoteAssetUrl(
+  normalizedBaseUrl: string,
+  assetPath: string,
+) {
+  // e.g. `https://localhost:3030/static/foo.js`
+  if (assetPath.startsWith('http')) {
```

Comment:
Personally I'd prefer these as if{} else if{} as only one can be true and we always return - make it clear that only one of those pieces of logic can run